Error compiling with boost asio included before TcpLayer.h

A simple program fails to compile
```cpp
#include <boost/asio.hpp>
#include <TcpLayer.h>

int main() {
  return 0;
}
```
Error is
```
/home/krepver/Dev/lib/PcapPlusPlus-master/Dist/header/TcpLayer.h:88:3: error: expected identifier before numeric constant
   TCPOPT_NOP =    1,
   ^
```

Boost ends up including <netinet/tcp.h>, which has line 163 `# define TCPOPT_NOP		1`, so TCPOPT_NOP in TcpOption enum is replaced by "1".

My system is Ubuntu 16.04, gcc 4.9, boost 1.55

--- a/Packet++/header/TcpLayer.h
+++ b/Packet++/header/TcpLayer.h
@@ -85,23 +85,23 @@ namespace pcpp
 	 */
 	enum TcpOption {
 		/** Padding */
-		TCPOPT_NOP = 			1,
+		PCPP_TCPOPT_NOP = 			1,
 		/** End of options */
-		TCPOPT_EOL = 			0,
+		PCPP_TCPOPT_EOL = 			0,
 		/** Segment size negotiating */
 		TCPOPT_MSS = 			2,
 		/** Window scaling */
-		TCPOPT_WINDOW = 		3,
+		PCPP_TCPOPT_WINDOW = 		3,
 		/** SACK Permitted */
 		TCPOPT_SACK_PERM = 		4,
 		/** SACK Block */
-		TCPOPT_SACK =           5,
-		/** Echo (obsoleted by option ::TCPOPT_TIMESTAMP) */
+		PCPP_TCPOPT_SACK =           5,
+		/** Echo (obsoleted by option ::PCPP_TCPOPT_TIMESTAMP) */
 		TCPOPT_ECHO =           6,
-		/** Echo Reply (obsoleted by option ::TCPOPT_TIMESTAMP) */
+		/** Echo Reply (obsoleted by option ::PCPP_TCPOPT_TIMESTAMP) */
 		TCPOPT_ECHOREPLY =      7,
 		/** TCP Timestamps */
-		TCPOPT_TIMESTAMP =      8,
+		PCPP_TCPOPT_TIMESTAMP =      8,
 		/** CC (obsolete) */
 		TCPOPT_CC =             11,
 		/** CC.NEW (obsolete) */
@@ -137,23 +137,23 @@ namespace pcpp
 
 	// TCP option lengths
 
-	/** pcpp::TCPOPT_NOP length */
+	/** pcpp::PCPP_TCPOPT_NOP length */
 #define PCPP_TCPOLEN_NOP            1
-	/** pcpp::TCPOPT_EOL length */
+	/** pcpp::PCPP_TCPOPT_EOL length */
 #define PCPP_TCPOLEN_EOL            1
 	/** pcpp::TCPOPT_MSS length */
 #define PCPP_TCPOLEN_MSS            4
-	/** pcpp::TCPOPT_WINDOW length */
+	/** pcpp::PCPP_TCPOPT_WINDOW length */
 #define PCPP_TCPOLEN_WINDOW         3
 	/** pcpp::TCPOPT_SACK_PERM length */
 #define PCPP_TCPOLEN_SACK_PERM      2
-	/** pcpp::TCPOPT_SACK length */
+	/** pcpp::PCPP_TCPOPT_SACK length */
 #define PCPP_TCPOLEN_SACK_MIN       2
 	/** pcpp::TCPOPT_ECHO length */
 #define PCPP_TCPOLEN_ECHO           6
 	/** pcpp::TCPOPT_ECHOREPLY length */
 #define PCPP_TCPOLEN_ECHOREPLY      6
-	/** pcpp::TCPOPT_TIMESTAMP length */
+	/** pcpp::PCPP_TCPOPT_TIMESTAMP length */
 #define PCPP_TCPOLEN_TIMESTAMP     10
 	/** pcpp::TCPOPT_CC length */
 #define PCPP_TCPOLEN_CC             6
@@ -240,7 +240,7 @@ namespace pcpp
 		 */
 		inline size_t getTotalSize() const
 		{
-			if (option == (uint8_t)TCPOPT_NOP || option == (uint8_t)TCPOPT_EOL)
+			if (option == (uint8_t)PCPP_TCPOPT_NOP || option == (uint8_t)PCPP_TCPOPT_EOL)
 				return sizeof(uint8_t);
 
 			return (size_t)len;

--- a/Packet++/src/TcpLayer.cpp
+++ b/Packet++/src/TcpLayer.cpp
@@ -151,7 +151,7 @@ bool TcpLayer::removeAllTcpOptions()
 
 TcpOptionData* TcpLayer::addTcpOptionAt(TcpOption optionType, uint8_t optionLength, const uint8_t* optionData, int offset)
 {
-	if ((optionType == TCPOPT_EOL || optionType == TCPOPT_NOP) && optionLength != PCPP_TCPOLEN_NOP)
+	if ((optionType == PCPP_TCPOPT_EOL || optionType == PCPP_TCPOPT_NOP) && optionLength != PCPP_TCPOLEN_NOP)
 	{
 		LOG_ERROR("Can't set TCP NOP option or TCP EOL option with size different than 1, tried to set size %d", optionLength);
 		return NULL;

--- a/Packet++Test/main.cpp
+++ b/Packet++Test/main.cpp
@@ -520,8 +520,8 @@ PACKETPP_TEST(TcpPacketNoOptionsParsing)
 
 	// TCP options
 	PACKETPP_ASSERT(tcpLayer->getTcpOptionsCount() == 0, "TCP options count isn't 0, it's %d", tcpLayer->getTcpOptionsCount());
-	PACKETPP_ASSERT(tcpLayer->getTcpOptionData(TCPOPT_NOP) == NULL, "TCP option NOP isn't NULL");
-	PACKETPP_ASSERT(tcpLayer->getTcpOptionData(TCPOPT_TIMESTAMP) == NULL, "TCP option Timestamp isn't NULL");
+	PACKETPP_ASSERT(tcpLayer->getTcpOptionData(PCPP_TCPOPT_NOP) == NULL, "TCP option NOP isn't NULL");
+	PACKETPP_ASSERT(tcpLayer->getTcpOptionData(PCPP_TCPOPT_TIMESTAMP) == NULL, "TCP option Timestamp isn't NULL");
 
 	Layer* afterTcpLayer = tcpLayer->getNextLayer();
 	PACKETPP_ASSERT(afterTcpLayer != NULL, "Layer after TCP is NULL");
@@ -557,8 +557,8 @@ PACKETPP_TEST(TcpPacketWithOptionsParsing)
 	PACKETPP_ASSERT(tcpLayer->getTcpOptionsCount() == 3, "TCP options count != 3, it's %d", tcpLayer->getTcpOptionsCount());
 	TcpOptionData* nopOptionData = NULL;
 	TcpOptionData* timestampOptionData = NULL;
-	PACKETPP_ASSERT((timestampOptionData = tcpLayer->getTcpOptionData(TCPOPT_TIMESTAMP)) != NULL, "TCP option Timestamp is NULL");
-	PACKETPP_ASSERT((nopOptionData = tcpLayer->getTcpOptionData(TCPOPT_NOP)) != NULL, "TCP option NOP is NULL");
+	PACKETPP_ASSERT((timestampOptionData = tcpLayer->getTcpOptionData(PCPP_TCPOPT_TIMESTAMP)) != NULL, "TCP option Timestamp is NULL");
+	PACKETPP_ASSERT((nopOptionData = tcpLayer->getTcpOptionData(PCPP_TCPOPT_NOP)) != NULL, "TCP option NOP is NULL");
 	PACKETPP_ASSERT(timestampOptionData->len == 10, "TCP option Timestamp length != 10, it's 0x%X", timestampOptionData->len);
 	uint32_t tsValue = 0;
 	uint32_t tsEchoReply = 0;
@@ -591,11 +591,11 @@ PACKETPP_TEST(TcpPacketWithOptionsParsing2)
 	TcpOptionData* windowScaleOptionData = NULL;
 	PACKETPP_ASSERT((mssOptionData = tcpLayer->getTcpOptionData(TCPOPT_MSS)) != NULL, "TCP option MSS is NULL");
 	PACKETPP_ASSERT((sackParmOptionData = tcpLayer->getTcpOptionData(TCPOPT_SACK_PERM)) != NULL, "TCP option SACK perm is NULL");
-	PACKETPP_ASSERT((windowScaleOptionData = tcpLayer->getTcpOptionData(TCPOPT_WINDOW)) != NULL, "TCP option window scale is NULL");
+	PACKETPP_ASSERT((windowScaleOptionData = tcpLayer->getTcpOptionData(PCPP_TCPOPT_WINDOW)) != NULL, "TCP option window scale is NULL");
 
 	PACKETPP_ASSERT(mssOptionData->getType() == TCPOPT_MSS, "MSS option isn't of type TCPOPT_MSS");
 	PACKETPP_ASSERT(sackParmOptionData->getType() == TCPOPT_SACK_PERM, "Sack perm option isn't of type TCPOPT_SACK_PERM");
-	PACKETPP_ASSERT(windowScaleOptionData->getType() == TCPOPT_WINDOW, "Window scale option isn't of type TCPOPT_WINDOW");
+	PACKETPP_ASSERT(windowScaleOptionData->getType() == PCPP_TCPOPT_WINDOW, "Window scale option isn't of type PCPP_TCPOPT_WINDOW");
 
 	PACKETPP_ASSERT(mssOptionData->len == 4, "TCP option Timestamp length != 4, it's 0x%X", mssOptionData->len);
 	PACKETPP_ASSERT(sackParmOptionData->len == 2, "TCP option SACK perm length != 2, it's 0x%X", sackParmOptionData->len);
@@ -612,11 +612,11 @@ PACKETPP_TEST(TcpPacketWithOptionsParsing2)
 	curOpt = tcpLayer->getNextTcpOptionData(curOpt);
 	PACKETPP_ASSERT(curOpt != NULL && curOpt->getType() == TCPOPT_SACK_PERM, "Second option isn't of type TCPOPT_SACK_PERM");
 	curOpt = tcpLayer->getNextTcpOptionData(curOpt);
-	PACKETPP_ASSERT(curOpt != NULL && curOpt->getType() == TCPOPT_TIMESTAMP, "Third option isn't of type TCPOPT_TIMESTAMP");
+	PACKETPP_ASSERT(curOpt != NULL && curOpt->getType() == PCPP_TCPOPT_TIMESTAMP, "Third option isn't of type PCPP_TCPOPT_TIMESTAMP");
 	curOpt = tcpLayer->getNextTcpOptionData(curOpt);
-	PACKETPP_ASSERT(curOpt != NULL && curOpt->getType() == TCPOPT_NOP, "Fourth option isn't of type TCPOPT_NOP");
+	PACKETPP_ASSERT(curOpt != NULL && curOpt->getType() == PCPP_TCPOPT_NOP, "Fourth option isn't of type PCPP_TCPOPT_NOP");
 	curOpt = tcpLayer->getNextTcpOptionData(curOpt);
-	PACKETPP_ASSERT(curOpt != NULL && curOpt->getType() == TCPOPT_WINDOW, "Fifth option isn't of type TCPOPT_WINDOW");
+	PACKETPP_ASSERT(curOpt != NULL && curOpt->getType() == PCPP_TCPOPT_WINDOW, "Fifth option isn't of type PCPP_TCPOPT_WINDOW");
 	curOpt = tcpLayer->getNextTcpOptionData(curOpt);
 	PACKETPP_ASSERT(curOpt == NULL, "There is sixth TCP option");
 
@@ -640,11 +640,11 @@ PACKETPP_TEST(TcpPacketCreation)
 	tcpLayer.getTcpHeader()->ackFlag = 1;
 	tcpLayer.getTcpHeader()->pshFlag = 1;
 	tcpLayer.getTcpHeader()->windowSize = htons(20178);
-	PACKETPP_ASSERT(tcpLayer.addTcpOption(TCPOPT_NOP, PCPP_TCPOLEN_NOP, NULL) != NULL, "Couldn't add 1st NOP option");
+	PACKETPP_ASSERT(tcpLayer.addTcpOption(PCPP_TCPOPT_NOP, PCPP_TCPOLEN_NOP, NULL) != NULL, "Couldn't add 1st NOP option");
 	PACKETPP_ASSERT(tcpLayer.getHeaderLen() == 24, "Header len isn't 24 after 1st NOP addition")
-	PACKETPP_ASSERT(tcpLayer.addTcpOption(TCPOPT_NOP, PCPP_TCPOLEN_NOP, NULL) != NULL, "Couldn't add 2nd NOP option");
+	PACKETPP_ASSERT(tcpLayer.addTcpOption(PCPP_TCPOPT_NOP, PCPP_TCPOLEN_NOP, NULL) != NULL, "Couldn't add 2nd NOP option");
 	PACKETPP_ASSERT(tcpLayer.getHeaderLen() == 24, "Header len isn't 24 after 2nd NOP addition")
-	TcpOptionData* tsOption = tcpLayer.addTcpOption(TCPOPT_TIMESTAMP, PCPP_TCPOLEN_TIMESTAMP, NULL);
+	TcpOptionData* tsOption = tcpLayer.addTcpOption(PCPP_TCPOPT_TIMESTAMP, PCPP_TCPOLEN_TIMESTAMP, NULL);
 	PACKETPP_ASSERT(tcpLayer.getHeaderLen() == 32, "Header len isn't 32 after timestamp addition")
 
 	PACKETPP_ASSERT(tsOption != NULL, "Couldn't add timestamp option");
@@ -662,7 +662,7 @@ PACKETPP_TEST(TcpPacketCreation)
 	tcpPacket.addLayer(&PayloadLayer);
 
 	uint32_t tsEchoReply = htonl(196757);
-	TcpOptionData* tsOptionData = tcpLayer.getTcpOptionData(TCPOPT_TIMESTAMP);
+	TcpOptionData* tsOptionData = tcpLayer.getTcpOptionData(PCPP_TCPOPT_TIMESTAMP);
 	PACKETPP_ASSERT(tsOptionData != NULL, "Couldn't get timestamp option");
 	tsOptionData->setValue<uint32_t>(tsEchoReply, 4);
 
@@ -711,7 +711,7 @@ PACKETPP_TEST(TcpPacketCreation2)
 	tcpLayer.getTcpHeader()->synFlag = 1;
 	tcpLayer.getTcpHeader()->windowSize = htons(14600);
 
-	TcpOptionData* nopOption = tcpLayer.addTcpOption(TCPOPT_NOP, PCPP_TCPOLEN_NOP, NULL);
+	TcpOptionData* nopOption = tcpLayer.addTcpOption(PCPP_TCPOPT_NOP, PCPP_TCPOLEN_NOP, NULL);
 	PACKETPP_ASSERT(nopOption != NULL, "Couldn't add NOP option");
 	PACKETPP_ASSERT(tcpLayer.getHeaderLen() == 24, "Header len isn't 24 after NOP addition");
 
@@ -720,13 +720,13 @@ PACKETPP_TEST(TcpPacketCreation2)
 	PACKETPP_ASSERT(mssOption != NULL, "Couldn't add MSS option");
 	PACKETPP_ASSERT(tcpLayer.getHeaderLen() == 28, "Header len isn't 28 after MSS addition")
 
-	TcpOptionData* tsOption = tcpLayer.addTcpOptionAfter(TCPOPT_TIMESTAMP, PCPP_TCPOLEN_TIMESTAMP, NULL, mssOption);
+	TcpOptionData* tsOption = tcpLayer.addTcpOptionAfter(PCPP_TCPOPT_TIMESTAMP, PCPP_TCPOLEN_TIMESTAMP, NULL, mssOption);
 	PACKETPP_ASSERT(tsOption != NULL, "Couldn't add timestamp option");
 	tsOption->setValue<uint32_t>(htonl(197364));
 	tsOption->setValue<uint32_t>(0, 4);
 	PACKETPP_ASSERT(tcpLayer.getHeaderLen() == 36, "Header len isn't 36 after timestamp addition")
 
-	TcpOptionData* winScaleOption = tcpLayer.addTcpOption(TCPOPT_WINDOW, PCPP_TCPOLEN_WINDOW, NULL);
+	TcpOptionData* winScaleOption = tcpLayer.addTcpOption(PCPP_TCPOPT_WINDOW, PCPP_TCPOLEN_WINDOW, NULL);
 	PACKETPP_ASSERT(winScaleOption != NULL, "Couldn't add Window Scale option");
 	winScaleOption->setValue<uint8_t>(4);
 	PACKETPP_ASSERT(tcpLayer.getHeaderLen() == 40, "Header len isn't 40 after Window scale addition");
@@ -771,15 +771,15 @@ PACKETPP_TEST(TcpPacketCreation2)
 	PACKETPP_ASSERT(tcpLayer.addTcpOptionAfter(TCPOPT_QS, PCPP_TCPOLEN_QS, (uint8_t*)&tcpQsValue, mssOption) != NULL, "Cannot add QS option");
 	int tcpSnackValue = htonl(1000);
 	PACKETPP_ASSERT(tcpLayer.addTcpOption(TCPOPT_SNACK, PCPP_TCPOLEN_SNACK, (uint8_t*)&tcpSnackValue) != NULL, "Cannot add SNACK option");
-	tsOption = tcpLayer.getTcpOptionData(TCPOPT_TIMESTAMP);
-	PACKETPP_ASSERT(tcpLayer.addTcpOptionAfter(TCPOPT_NOP, PCPP_TCPOLEN_NOP, NULL, tsOption) != NULL, "Cannot add 2nd NOP option");
+	tsOption = tcpLayer.getTcpOptionData(PCPP_TCPOPT_TIMESTAMP);
+	PACKETPP_ASSERT(tcpLayer.addTcpOptionAfter(PCPP_TCPOPT_NOP, PCPP_TCPOLEN_NOP, NULL, tsOption) != NULL, "Cannot add 2nd NOP option");
 
 	PACKETPP_ASSERT(tcpLayer.getTcpOptionsCount() == 8, "TCP option count isn't 8");
 
 	PACKETPP_ASSERT(tcpLayer.removeTcpOption(TCPOPT_QS) == true, "Cannot remove QS option");
 	PACKETPP_ASSERT(tcpLayer.getTcpOptionsCount() == 7, "TCP option count isn't 7");
 	PACKETPP_ASSERT(tcpLayer.removeTcpOption(TCPOPT_SNACK) == true, "Cannot remove SNACK option");
-	PACKETPP_ASSERT(tcpLayer.removeTcpOption(TCPOPT_NOP) == true, "Cannot remove NOP option");
+	PACKETPP_ASSERT(tcpLayer.removeTcpOption(PCPP_TCPOPT_NOP) == true, "Cannot remove NOP option");
 	PACKETPP_ASSERT(tcpLayer.getTcpOptionsCount() == 5, "TCP option count isn't 5 again");
 
 	PACKETPP_ASSERT(memcmp(tcpPacket.getRawPacket()->getRawData(), buffer, bufferLength) == 0, "Raw packet data is different than expected");
@@ -790,7 +790,7 @@ PACKETPP_TEST(TcpPacketCreation2)
 	PACKETPP_ASSERT(tcpLayer.getTcpOptionsCount() == 0, "TCP option count isn't zero after removing all of them");
 	PACKETPP_ASSERT(tcpLayer.getFirstTcpOptionData() == NULL, "Found TCP option after removing all of them");
 	PACKETPP_ASSERT(tcpLayer.getHeaderLen() == 20, "Header len isn't 20 after removing all TCP options");
-	PACKETPP_ASSERT(tcpLayer.getTcpOptionData(TCPOPT_TIMESTAMP) == NULL, "Found TS option after removing all of TCP options");
+	PACKETPP_ASSERT(tcpLayer.getTcpOptionData(PCPP_TCPOPT_TIMESTAMP) == NULL, "Found TS option after removing all of TCP options");
 
 	PACKETPP_ASSERT(tcpLayer.addTcpOption(TCPOPT_SNACK, PCPP_TCPOLEN_SNACK, (uint8_t*)&tcpSnackValue) != NULL, "Cannot add SNACK option again");
 
@@ -2269,9 +2269,9 @@ PACKETPP_TEST(CopyLayerAndPacketTest)
 			"TcpLayer copy c'tor didn't copy data properly, original and copied data isn't equal");
 	PACKETPP_ASSERT(tcpLayer.getTcpOptionsCount() == sampleTcpPacketWithOptions.getLayerOfType<TcpLayer>()->getTcpOptionsCount(),
 			"TcpLayer copy and original TCP options count is not equal");
-	PACKETPP_ASSERT(sampleTcpPacketWithOptions.getLayerOfType<TcpLayer>()->getTcpOptionData(TCPOPT_TIMESTAMP) != tcpLayer.getTcpOptionData(TCPOPT_TIMESTAMP),
+	PACKETPP_ASSERT(sampleTcpPacketWithOptions.getLayerOfType<TcpLayer>()->getTcpOptionData(PCPP_TCPOPT_TIMESTAMP) != tcpLayer.getTcpOptionData(PCPP_TCPOPT_TIMESTAMP),
 			"TcpLayer copy and original TCP Timestamp option pointer is the same");
-	PACKETPP_ASSERT(memcmp(sampleTcpPacketWithOptions.getLayerOfType<TcpLayer>()->getTcpOptionData(TCPOPT_TIMESTAMP), tcpLayer.getTcpOptionData(TCPOPT_TIMESTAMP), PCPP_TCPOLEN_TIMESTAMP) == 0,
+	PACKETPP_ASSERT(memcmp(sampleTcpPacketWithOptions.getLayerOfType<TcpLayer>()->getTcpOptionData(PCPP_TCPOPT_TIMESTAMP), tcpLayer.getTcpOptionData(PCPP_TCPOPT_TIMESTAMP), PCPP_TCPOLEN_TIMESTAMP) == 0,
 			"TcpLayer copy and original TCP Timestamp option data differs");
 
 
@@ -3962,9 +3962,9 @@ PACKETPP_TEST(SllPacketCreationTest)
 	tcpLayer.getTcpHeader()->ackNumber = htonl(0x7633e977);
 	tcpLayer.getTcpHeader()->ackFlag = 1;
 	tcpLayer.getTcpHeader()->windowSize = htons(4098);
-	PACKETPP_ASSERT(tcpLayer.addTcpOption(TCPOPT_NOP, PCPP_TCPOLEN_NOP, NULL) != NULL, "Cannot add 1st NOP option");
-	PACKETPP_ASSERT(tcpLayer.addTcpOption(TCPOPT_NOP, PCPP_TCPOLEN_NOP, NULL) != NULL, "Cannot add 2nd NOP option");
-	TcpOptionData* tsOption = tcpLayer.addTcpOption(TCPOPT_TIMESTAMP, PCPP_TCPOLEN_TIMESTAMP, NULL);
+	PACKETPP_ASSERT(tcpLayer.addTcpOption(PCPP_TCPOPT_NOP, PCPP_TCPOLEN_NOP, NULL) != NULL, "Cannot add 1st NOP option");
+	PACKETPP_ASSERT(tcpLayer.addTcpOption(PCPP_TCPOPT_NOP, PCPP_TCPOLEN_NOP, NULL) != NULL, "Cannot add 2nd NOP option");
+	TcpOptionData* tsOption = tcpLayer.addTcpOption(PCPP_TCPOPT_TIMESTAMP, PCPP_TCPOLEN_TIMESTAMP, NULL);
 	PACKETPP_ASSERT(tsOption != NULL, "Couldn't set timestamp TCP option");
 	tsOption->setValue<uint32_t>(htonl(0x0402383b));
 	tsOption->setValue<uint32_t>(htonl(0x03ff37f5), 4);

